assets/react/main.js

```diff
@@ -3,55 +3,56 @@ var Router = require('react-router');
 var Route = Router.Route;
 var Wrapper = require('./wrapper.js');
 var Home = require('./components/home.js');
-var MyPosts = require('./components/my_posts.js');
+var Profile = require('./components/profile.js');
 var EditPost = require('./components/edit_post.js');
 var ShowPost = require('./components/show_post.js');
 import {Filters} from './constants/visibilityFilters';
-import {fetchAllPosts} from './actions/posts';
 import { Provider } from 'react-redux';
 import configureStore from './store/store';
 import { connect } from 'react-redux';
-import {createPosts, setSelectedPost, setSelectedPostByName, setSelectedPostById , requestPost, fetchMyPosts} from './actions/posts';
+import {createPosts, setSelectedPost, setSelectedPostByName, setSelectedPostById , requestPost, fetchProfile, fetchPosts} from './actions/posts';
 import * as VisibilityConstants from './constants/visibilityFilters'
-import {setUserData} from './actions/user'
+import {addUserData, setCurrentUserId, getUserData, setProfileId} from './actions/user'
 import {setVisibilityFilter} from './actions/visibilityFilters'
 
 
-//Elements generated with wrapper
-// function getInitialData(){
-// 	var initialState = window.fodoo_data
-// 	return {
-// 		user: initialState.user,
-// 		posts: initialState.posts
-// 	}
-// }
 const store = configureStore();
 
 store.dispatch(createPosts(window.fodoo_data.posts));
-store.dispatch(setUserData(window.fodoo_data.user));
+store.dispatch(setCurrentUserId(window.fodoo_data.user && window.fodoo_data.user.id));
+store.dispatch(addUserData(window.fodoo_data.user));
+store.dispatch(addUserData(window.fodoo_data.profile_user));
```

Comment:
Why do we need a seperate dispatch for profile user? Both current user and profile user can be received in one array.